java/IBAutomater/src/ibautomater/WindowEventListener.java

```diff
@@ -259,6 +271,110 @@ private boolean HandleLoginFailedWindow(Window window, int eventId) {
         return false;
     }
 
+    private boolean HandleServerDisconnectedWindow(Window window, int eventId) {
+        if (eventId != WindowEvent.WINDOW_OPENED) {
+            return false;
+        }
+
+        String text = GetWindowText(window);
+
+        if (text != null && text.contains("Connection to server failed: Server disconnected, please try again")) {
+            this.automater.logMessage(text);
+
+            if (IsWithinWeekendServerResetTimes())
+            {
+                this.automater.logMessage("Server disconnection detected during weekend server reset times, delaying the reconnection attempt.");
+
+                // start thread to wait until one hour before FX market open before retrying login
+                new Thread(()-> {
+                    try {
+                        Duration delta = Duration.between(Instant.now(), GetNextWeekendReconnectionTimeUtc());
+                        long delay = delta.getSeconds() * 1000;
+
+                        Thread.sleep(delay);
+
+                        // execute asynchronously on the AWT event dispatching thread
+                        SwingUtilities.invokeLater(() -> {
+                            try {
+                                JButton button = Common.getButton(window, "OK");
+                                if (button != null) {
+                                    this.automater.logMessage("Click button: [OK]");
+                                    button.doClick();
+                                }
+
+                                Window mainWindow = automater.getMainWindow();
+                                HandleLoginWindow(mainWindow, WindowEvent.WINDOW_OPENED);
+                            } catch (Exception e) {
+                                automater.logMessage("HandleLoginWindow error: " + e.getMessage());
+                            }
+                        });
+                    } catch (Exception e) {
+                        automater.logMessage("HandleLoginWindow error: " + e.getMessage());
+                    }
+                }).start();
+            }
```

Comment:
What will happen with the `else` case here, say we are not within weekend server reset times?

Reply:
From what we've seen in the past this has only happened during the weekend,
if not we would just log the message and timeout.